psutil/arch/windows/process_info.c

```diff
@@ -687,16 +671,86 @@ static int psutil_get_parameters(long pid, PyObject **pcmdline,
 PyObject *
 psutil_get_cmdline(long pid) {
     PyObject *ret = NULL;
-    psutil_get_parameters(pid, &ret, NULL);
+    WCHAR *data = NULL;
+    SIZE_T size;
+    PyObject *py_retlist = NULL;
+    PyObject *py_unicode = NULL;
+    LPWSTR *szArglist = NULL;
+    int nArgs, i;
+
+    if (psutil_get_process_data(pid, KIND_CMDLINE, &data, &size) != 0)
+        goto out;
+
+    // attempt to parse the command line using Win32 API
+    szArglist = CommandLineToArgvW(data, &nArgs);
+    if (szArglist == NULL) {
+        PyErr_SetFromWindowsErr(0);
+        goto out;
+    }
+
+    // arglist parsed as array of UNICODE_STRING, so convert each to
+    // Python string object and add to arg list
+    py_retlist = PyList_New(nArgs);
+    if (py_retlist == NULL)
+        goto out;
+    for (i = 0; i < nArgs; i++) {
+        py_unicode = PyUnicode_FromWideChar(
+                szArglist[i], wcslen(szArglist[i]));
+        if (py_unicode == NULL)
+            goto out;
+        PyList_SET_ITEM(py_retlist, i, py_unicode);
```

Comment:
Hm.  `PyList_Append` seems pointless since we already know the correct size.  `PyList_SetItem`/`PyList_SET_ITEM` steal so there's no need to `Py_DECREF`.  Anyway, I'll revert it if you want since I changed for no real reason.